src/hotspot/share/runtime/perfData.hpp

```diff
@@ -831,11 +831,17 @@ class PerfTraceTime : public StackObj {
 
   public:
     inline PerfTraceTime(PerfLongCounter* timerp) : _timerp(timerp) {
-      if (!UsePerfData) return;
+      if (!UsePerfData || timerp == nullptr) return;
```

Comment:
Okay so this is needed because the existence of some counters is gated on the ProfileClassLinkage flag.

Style nit: use a { } block please.

Reply:
Will fix.